packages/snjs/lib/models/app/search.ts

```diff
@@ -0,0 +1,40 @@
+/** Grouping of required state variables to perform the search */
+
+import { CollectionSort } from "@Lib/protocol/collection/item_collection";
+import { SNTag } from "./tag";
+
+export interface SearchPayload {
+  tag: SNTag,
+  sortBy: CollectionSort,
+  isReversedSort: "asc" | "dsc" | undefined,
+  showArchiveOrTrashed: boolean,
+  hidePinned: boolean,
+  query: string
+
+}
+
+export class Search {
+
+  /**
+   * This is a static class so instantiation should be impossible.
+   */
+  private constructor() {
```

Comment:
Can a constructor be private?

Reply:
Yeah! It's quite cool. Allows 'simulating' a static class.
Allows forcing Name.method() style coding instead of just function().